clients/include/testing_matmul.hpp

```diff
@@ -3257,11 +3271,22 @@ void testing_matmul_with_bias(const Arguments& arg,
             int32_t     solutionIndex = -1;
             std::string solutionName  = "";
             std::string kernelName    = "";
-            if(arg.print_kernel_info)
+            std::string archName      = "";
+            std::string cuNum         = "";
+            const char* tuningEnv     = getenv("HIPBLASLT_TUNING_FILE");
+            if(tuningEnv)
             {
                 solutionIndex = hipblaslt_ext::getIndexFromAlgo(heuristicResult[best_sol].algo);
-                solutionName  = best_s_name;
-                kernelName    = best_k_name;
+                archName      = deviceProps.gcnArchName;
+                cuNum         = std::to_string(deviceProps.multiProcessorCount);
+            }
+
```

Comment:
if(tuningEnv ||arg.print_kernel_info)
getindex
if(tuningEnv)
get archName and CuNum
if(arg.print_kernel_info)
get solutionname and kernel name

Reply:
Thanks, it’s done in commit [28cfcc6](https://github.com/ROCm/hipBLASLt/pull/1289/commits/28cfcc6c7dff7b30e7b139f6db043d8ae3e44131)